### ISSUE TITLE
incomplete pod2html/Pod::Html documentation on --htmldir 

### ISSUE THREAD

**Issue by pertusus** [Perl/perl5#20527]
**Where**
Pod::Html and pod2html

**Description**
The documentation of pod2html/Pod::Html seems to me to be incomplete about --htmldir.  Indeed, it is not said that --outfile is used to set the relative path and that the leading part of outfile is compared to --htmldir to set the relative path.

I had to read the source (in refine_globals) to get it right.

A possibility (I am not a native english speaker) for =item htmldir:

***

The beginning of --outfile should match with --htmldir to actually use a relative directory, in order to determine the relative path.

***

Two examples of calls that behave as if --htmldir was not given without an indication in the documentation that --htmldir would be ignored in those cases:

 `pod2html --htmldir='.' --podpath=. my_infile.pm > result.html`

 `pod2html --htmldir='.' --podpath=. my_infile.pm --outfile=result.html`

The call that worked as intended

 `pod2html --htmldir='.' --podpath=. my_infile.pm --outfile=./result.html`

**Comment by jkeenan** [comment]
> **Description** The documentation of pod2html/Pod::Html seems to me to be incomplete about `--htmldir`. Indeed, it is not said that `--outfile` is used to set the relative path and that the leading part of outfile is compared to `--htmldir` to set the relative path.
> 

Can we ask you to provide a **small** example of the actual problem you are trying to solve, during which you had difficulty understanding the documentation?

Since your problem refers to *relative* paths, my hunch is that you have more than one file, each of which contains documentation in POD format, and at least one of which contains a crosslink to another file, *e.g.,* `L<crosslink_to_file.pod>`.  This is my hunch because, in my experience, the `--htmldir` argument is largely irrelevant unless you are preparing to host a tree of web pages which have to be able to find one another on a given server.

So, if what I have just described is your practical problem, giving us two files with POD which crosslink to each other would be helpful.

On the other hand, if your problem is different from what I have guessed, please provide a *small* file which illustrates the problem.  Thanks for your report.

**Comment by pertusus** [comment]
Indeed, it is as you say, although it is not cross-links, only one link. I have one file with pod that links to another, and I want to be able to convert both files as html with the link working correctly when uploaded to a website. The file with pod linking to the other file is named pod2texi.pl. The file it links to is Pod/Simple/Texinfo.pm. In pod2texi.pl there is, in the SEE ALSO section `L<Pod::Simple::Texinfo>`.

`$ ls Pod/Simple/Texinfo.pm`
`Pod/Simple/Texinfo.pm`
The actual call that works is
`pod2html --podpath . --htmldir='.' --quiet /home/dumas/src/texinfo/Pod-Simple-Texinfo/pod2texi.pl --outfile=./pod2texi.html`
The ones that did not work, but there is nothing in the documentation explaining that they will fail because the outfile beginning does not match the htmldir are:
`pod2html --podpath . --htmldir='.' --quiet /home/dumas/src/texinfo/Pod-Simple-Texinfo/pod2texi.pl > pod2texi.html`
`pod2html --podpath . --htmldir='.' --quiet /home/dumas/src/texinfo/Pod-Simple-Texinfo/pod2texi.pl --outfile=pod2texi.html`

The result is actually on the web:
https://www.gnu.org/software/texinfo/manual/pod2texi.html
and the link target:
https://www.gnu.org/software/texinfo/manual/Pod/Simple/Texinfo.html

[files_pod.tar.gz](https://github.com/Perl/perl5/files/10048089/files_pod.tar.gz)

**Comment by pertusus** [comment]
My call to pod2html to create the other file is simpler, as there is no links I want to have resolved:
`pod2html --quiet Pod/Simple/Texinfo.pm >Pod/Simple/Texinfo.html`
